core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -56,6 +57,37 @@
     @ConfigItem
     public Optional<List<String>> userConfiguredIgnoredEntries;
 
+    /**
+     * List of all the dependencies that have been defined as optional to include into the final package of the application.
+     * Each optional dependency needs to be expressed in the following format:
+     * <p>
+     * groupId:artifactId:classifier:type
+     * <p>
+     * With the classifier and type being optional.
+     * <p>
+     * If the type is missing, the artifact is assumed to be of type {@code jar}.
+     * <p>
+     * This parameter is optional, if absent, no optional dependencies will be included into the final package of
+     * the application.
+     * <p>
+     * For backward compatibility reasons, this parameter is ignored by default and can be enabled by setting the
+     * parameter {@code quarkus.package.filter-optional-dependencies} to {@code true}.
+     * <p>
+     * This parameter is meant to be used in modules where multi-builds have been configured to avoid getting a final
+     * package with unused dependencies.
+     */
+    @ConfigItem
+    public Optional<Set<String>> optionalDependencies;
```

Comment:
IMO, "optional" dependencies is an unfortunate term. It's not really intuitive what that actually means. IMO, `excludedDependencies` or `ignoredDependencies` (similar to `userConfiguredIgnoredEntries`) would be better. WDYT?

Reply:
Or, perhaps, to be even more precise `ignoredOptionalDependencies`? As it looks like the filtering is only applied to the Maven optional deps.

Reply:
Ah, sorry, this is configuring what should be included.

Reply:
`includedOptionalDependencies`?